### PR TITLE
remove mentions of old versions of Nomad in various docs

### PR DESCRIPTION
**Pull request by sofixa**

### CODE REVIEW

File: website/content/docs/concepts/variables.mdx [hunk 1]
```diff
@@ -74,7 +74,7 @@ The available capabilities for Variables are as follows:
 
 ## Task Access to Variables
 
-In Nomad 1.4.0 tasks can only access Variables with the [`template`] block. The
+Tasks can only access Variables with the [`template`] block. The
```
Comment (tgross) [inline website/content/docs/concepts/variables.mdx:77]: Removing the version here changes the meaning of "only" here in a way that's no longer true. In Nomad 1.5.0+, tasks can also access Variables via the Task API.

File: website/content/api-docs/json-jobs.mdx [hunk 2]
```diff
@@ -616,9 +614,8 @@ The `Task` object supports the following keys:
     - `AddressMode`: Same as `AddressMode` on `Service`. Unlike services,
       checks do not have an `auto` address mode as there's no way for
       Nomad to know which is the best address to use for checks. Consul
-      needs access to the address for any HTTP or TCP checks. Added in
-      Nomad 0.7.1. Unlike `PortLabel`, this setting is _not_ inherited
-      from the `Service`.
+      needs access to the address for any HTTP or TCP checks. Unlike `PortLabel`,
+       this setting is _not_ inherited from the `Service`.
```
Comment (tgross) [inline website/content/api-docs/json-jobs.mdx:618]:
```suggestion
      this setting is _not_ inherited from the `Service`.
```

File: website/content/api-docs/json-jobs.mdx [hunk 3]
```diff
@@ -1099,10 +1095,9 @@ creates a convenient way to ship configuration files that are populated from
 environment variables, Consul data, Vault secrets, or just general
 configurations within a Nomad task.
 
-Nomad utilizes a tool called [Consul Template][ct]. Since Nomad v0.5.3, the
-template can reference [Nomad's runtime environment variables][env]. For a full
-list of the API template functions, please refer to the [Consul Template
-README][ct].
+Nomad utilizes a tool called [Consul Template][ct]. The template can reference 
+[Nomad's runtime environment variables][env]. For a full list of the API template
+ functions, please refer to the [Consul Template README][ct].
```
Comment (tgross) [inline website/content/api-docs/json-jobs.mdx:1100]:
```suggestion
functions, please refer to the [Consul Template README][ct].
```

File: website/content/docs/drivers/index.mdx [hunk 4]
```diff
@@ -10,11 +10,10 @@ Task drivers are used by Nomad clients to execute a task and provide resource
...
 
-Starting with Nomad 0.9, task drivers are now pluggable. This gives users the
-flexibility to introduce their own drivers without having to recompile Nomad.
-You can view the [plugin block][plugin] documentation for examples on how to
-use the `plugin` block in Nomad's client configuration. Note that we have
-introduced new syntax when specifying driver options in the client configuration
+Task drivers are pluggable. This gives users the flexibility to introduce their
+own drivers without having to recompile Nomad. You can view the [plugin block][plugin]
+documentation for examples on how to use the `plugin` block in Nomad's client configuration.
+Note that we have introduced new syntax when specifying driver options in the client configuration
```
Comment (tgross) [inline website/content/docs/drivers/index.mdx:16]: "we have introduced new syntax" is also version specific to 0.9. We should clean this sentence up too.

File: website/content/docs/job-specification/service.mdx [hunk 5]
```diff
@@ -70,9 +70,9 @@ This section of the documentation only cover the job file fields and blocks
 for service discovery. For more details on using Nomad with Consul please see
 the [Consul integration documentation][service-discovery].
 
-Nomad 0.10 also allows specifying the `service` block at the task group level.
+The `service` block can also be at task group level.
 This enables services in the same task group to opt into [Consul
-Connect][connect] integration.
+Service Mesh][connect] integration.
```
Comment (tgross) [inline website/content/docs/job-specification/service.mdx:75]: Did this get re-branded? I think we have "Connect" all over the place in the Nomad docs.

Comment (sofixa) [reply website/content/docs/job-specification/service.mdx:75]: Yep, it's "Consul Service Mesh" now. I know that Nomad mostly refers to it as Connect (including the job spec itself), but googling for "Nomad Consul Connect" results in those two high level docs first, both of which use "Service Mesh":

https://developer.hashicorp.com/nomad/docs/integrations/consul/service-mesh

https://developer.hashicorp.com/consul/docs/connect/nomad

So I think at least _some_ of the high level docs in Nomad land have started getting updated to the new wording, and opportunistically I updated this one too.

File: website/content/docs/concepts/variables.mdx (hunk 1 above)
Comment (sofixa) [reply website/content/docs/concepts/variables.mdx]: Oh true, I forgot that one, fixed

File: website/content/docs/drivers/index.mdx (hunk 4 above)
Comment (sofixa) [reply website/content/docs/drivers/index.mdx]: Cleaned up and clarified in the latest commit

File: website/content/docs/concepts/variables.mdx [hunk 6]
```diff
@@ -248,3 +248,4 @@ implementation or the [Nomad Autoscaler][] High Availability implementation.
 [Go Package]: https://pkg.go.dev/github.com/hashicorp/nomad/api
 [implementation]: https://github.com/hashicorp/nomad/blob/release/1.7.0/command/var_lock.go#L240
 [Nomad Autoscaler]: https://github.com/hashicorp/nomad-autoscaler/blob/v0.4.0/command/agent.go#L392
+[Task API]: nomad/api-docs/task-api
```
Comment (tgross) [inline website/content/docs/concepts/variables.mdx:251]:
```suggestion
[Task API]: /nomad/api-docs/task-api
```

File: website/content/docs/job-specification/index.mdx [hunk 7]
```diff
@@ -132,9 +132,9 @@ job "docs" {
 }
 ```
 
-Note that starting with Nomad 0.10, the `service` block can also be specified at the group level. This
-allows job specification authors to create and register services with Consul Connect support. A service
-block specified at the group level must include a [connect][] block, like the following snippet.
+The `service` block can also be specified at the group level. This allows job specification authors
+to create and register services with Consul Service Mesh. A service block specified at the group 
+level must include a [connect][] block, like the following snippet.
```
Comment (tgross) [inline website/content/docs/job-specification/index.mdx:137]:
```suggestion
level must include a [`connect`][] block, like the following snippet.
```

**Review by tgross — CHANGES_REQUESTED** [review]
Couple minor syntax bits and we should be good-to-merge

**Review by tgross — APPROVED** [review]
LGTM!

**Pull request merged**